libnavigation-core/src/test/java/com/mapbox/navigation/core/fasterroute/FasterRouteControllerTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.navigation.core.directions.session.AdjustedRouteOptionsProvider
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.testing.MainCoroutineRule
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import io.mockk.verify
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class FasterRouteControllerTest {
+
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    private val directionsSession: DirectionsSession = mockk()
+    private val tripSession: TripSession = mockk()
+    private val fasterRouteObserver: FasterRouteObserver = mockk()
+    private val routesRequestCallbacks = mutableListOf<RoutesRequestCallback>()
+
+    private val fasterRouteController = FasterRouteController(directionsSession, tripSession)
+
+    @Before
+    fun setup() {
+        mockkObject(AdjustedRouteOptionsProvider)
+        every { AdjustedRouteOptionsProvider.getRouteOptions(any(), any(), any()) } returns mockk()
+
+        every { directionsSession.getRouteOptions() } returns mockk()
+        every { directionsSession.requestFasterRoute(any(), capture(routesRequestCallbacks)) } returns mockk()
+        every { tripSession.getRouteProgress() } returns mockk()
+    }
+
+    @After
+    fun teardown() {
+        unmockkObject(AdjustedRouteOptionsProvider)
+    }
+
+    @Test
+    fun `should request every minute`() = coroutineRule.runBlockingTest {
+        every { directionsSession.routes } returns listOf(
+            mockk {
+                every { routeIndex() } returns "0"
+                every { distance() } returns 100.0
+            }
+        )
+        every { tripSession.getEnhancedLocation() } returns mockk {
+            every { latitude } returns -33.874308
+            every { longitude } returns 151.206087
+        }
+
+        fasterRouteController.attach(fasterRouteObserver)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(4))
```

Comment:
i thought about making a test to ensure a route wasn't requested in the first minute. my feeling was that was double testing something, those tests are in MapboxTimerTest

Reply:
this test is really a positive test case (make sure the controller makes requests). we need a positive test to go with the failure test